src/roles/transporter.ts

```diff
@@ -0,0 +1,76 @@
+import { RoleName } from "enums/RoleName";
+import { Tasks } from "../creep-tasks/Tasks";
+import { Role } from "./Role";
+
+export class RoleHarvester extends Role {
+    public static roleName: string = RoleName.Harvester;
+
+    public newTask(creep: Creep): void {
+        if (creep.carry.energy < creep.carryCapacity) {
+            // Find Structures for getting energy
+            const structures = creep.room.find(FIND_STRUCTURES);
+
+            // Withdraw from containers
+            const containersWithEnergy = structures.filter(s => s.structureType === STRUCTURE_CONTAINER && s.store[RESOURCE_ENERGY] > 0) as StructureContainer[];
+            if (containersWithEnergy.length > 0) {
+                creep.task = Tasks.withdraw(containersWithEnergy[0], RESOURCE_ENERGY);
+                return;
+            }
+
+            // Pickup dropped energy
+            const droppedEnergy = creep.pos.findClosestByPath(FIND_DROPPED_RESOURCES, { filter: (d) => d.resourceType === RESOURCE_ENERGY });
```

Comment:
Personally I think dropped energy isn't worth the time/effort, but this is something we can discuss, more of an opinion.

Reply:
We can talk about it.